### PR TITLE
Revert "Update IERS_B as well as its URL"

### PR DESCRIPTION
**Pull request by pllim**
Reverts astropy/astropy#14384

### CODE REVIEW

File: astropy/utils/iers/iers.py [hunk 1]
```diff
@@ -69,7 +69,7 @@
 
 # IERS-B default file name, URL, and ReadMe with content description
 IERS_B_FILE = get_pkg_data_filename("data/eopc04_IAU2000.62-now")
-IERS_B_URL = "https://hpiers.obspm.fr/iers/eop/eopc04_14/eopc04_IAU2000.62-now"
+IERS_B_URL = "http://hpiers.obspm.fr/iers/eop/eopc04/eopc04_IAU2000.62-now"
```
Comment (pllim) [inline astropy/utils/iers/iers.py:72]: Maybe we keep the https?

**Comment by pllim** [comment]
Actually, should we also keep the new rows from the file? Maybe we shouldn't do a full revert if that is the case. @mhvk ?

**Comment by pllim** [comment]
The mpldev failure makes no sense. When I click Details, everything is green. I will just ignore it.

**Comment by mhvk** [comment]
@pllim - yes, you're right, might as well keep the update to the IERS table. I guess that means only changing back the URLs.

**Comment by pllim** [comment]
OK I'll do that from my fork then.

**Pull request closed without merging**